test/governance/candidacy.users.test.ts

```diff
@@ -0,0 +1,233 @@
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo wins the election and is council member!
+    // Gonzalo subsequently renounces candidacy in Term 1
+    await waitForNextTerm();
+
+    const termOne = getLastBlocksElectionData();
+    expect(flattenArray(termOne.Members)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    const candidateStatus = { Member: candidate.keyRingPair.address };
+    await candidate.renounceCandidacy(candidateStatus);
+
+    // Gonzalo is not automatically running for candidacy in Term 2
+    await waitForNextTerm();
+
+    const termTwo = getLastBlocksElectionData();
+
+    expect(flattenArray(termTwo.Candidates)).not.toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+
+  test("Council Member automatically runs for candidacy again", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    // Gonzalo runs for Council in Term 0
+    // Eddy votes for Gonzalo
+    await candidate.runForCouncil();
+    await voter.vote([candidate], stakeAmount);
+
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+    expect(flattenArray(termZero.Candidates)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo wins the election and is a council member!
+    await waitForNextTerm();
+
+    const termOne = getLastBlocksElectionData();
+
+    expect(flattenArray(termOne.Members)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo is automatically running for candidacy in Term 2
+    await waitForNextTerm();
+
+    const termTwo = getLastBlocksElectionData();
+
+    expect(flattenArray(termTwo.Candidates)).not.toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+});
+
+async function waitForNextTerm() {
+  // Blocks per term are 50 on the modified mangata node
+  // We know the last block, so we can work out our position in the term
+  // as lastBlock % blocksPerTerm. We wait an additional 3 blocks purely to avoid
+  // tests needing to manually wait a few extra blocks too.
+  const blocksPerTerm = termDuration;
+  const lastBlock = bootnode.lastBlock!;
+  const blocksToWait =
+    blocksPerTerm - (lastBlock % blocksPerTerm) + termBlockDelta;
+  await waitForNBlocks(blocksToWait);
+}
+
+function getLastBlocksElectionData() {
+  const electionData = bootnode.electionEvents.get(bootnode.lastBlock! - 1);
+
+  return {
+    Members: electionData!.members,
+    Candidates: electionData!.candidates,
+  };
+}
+
+function flattenArray(arr: [any]) {
+  return arr.reduce((acc, val) => acc.concat(val), []);
+}
+
+async function cleanState() {
+  state.accounts.forEach(async (account) => {
+    if (account !== null) {
+      try {
+        const candidateStatus = { Member: account.keyRingPair.address };
+        await account?.renounceCandidacy(candidateStatus);
+      } catch (e) {} // This may fail if an account isn't a candidate
```

Comment:
will this clean the state if the account is a member?

Reply:
Added additional renounceCandidacy for members